db/import.go

```diff
@@ -78,17 +78,22 @@ func (db *Database) ImportDoc(docid string, existingDoc *Document, isDelete bool
 
 	// TODO: We need to remarshal the existing doc into bytes.  Less performance overhead than the previous bucket op to get the value in WriteUpdateWithXattr,
 	//       but should refactor import processing to support using the already-unmarshalled doc.
-	rawValue, rawXattr, err := existingDoc.MarshalWithXattr()
-	if err != nil {
-		return nil, err
-	}
+
 	existingBucketDoc := &sgbucket.BucketDocument{
-		Body:   rawValue,
-		Xattr:  rawXattr,
 		Cas:    existingDoc.Cas,
 		Expiry: *expiry,
 	}
 
+	if existingDoc.inlineSyncData {
+		existingBucketDoc.Body, err = existingDoc.MarshalJSON()
```

Comment:
unhandled error, maybe move the if `err != nil` check out of the else below to cover both?